BigBaseV2/src/util/models.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include "gta/joaat.hpp"
+#include "pointers.hpp"
+
+namespace big
+{
+	// iterate all models
+	/*
+	for (int i = 0; i < m_model_table->m_size; ++i)
+	{
+		for (auto node = m_model_table->m_lookup_table[i]; node; node = node->m_next)
+		{
+			if (const auto table_idx = node->m_idx; table_idx < m_model_table->m_size)
+			{
+				if (const auto model = m_model_table->m_data[table_idx]; model && model->m_model_type == eModelType::Vehicle)
+				{
+
+				}
+			}
+		}
+	}*/
+
+	class model_info
+	{
+	public:
+		static bool does_model_exist(const rage::joaat_t hash)
+		{
+			if (const auto model = model_info::get_model(hash); model)
+				return true;
+			return false;
+		}
+
+		template<typename T = CBaseModelInfo*>
+		static T get_model(const rage::joaat_t hash)
+		{
+			const auto model_table = g_pointers->m_model_table;
+			for (auto i = m_model_table->m_lookup_table[adder_hash % m_model_table->m_lookup_key]; i; i = i->m_next)
+			{
+				if (i->m_hash == hash)
+				{
+					if (const auto model = m_model_table->m_data[i->m_idx]; model)
+					{
+						return model;
+					}
+				}
+			}
+			return nullptr;
+		}
+
+		static CVehicleModelInfo* get_vehicle_model(const rage::joaat_t hash)
+		{
+			if (const auto model = model_info::get_model<CVehicleModelInfo*>(hash); model && model->m_model_type == eModelType::Vehicle)
```

Comment:
There are sub-types of Vehicle that need to be checked as well.

Reply:
I sent all of these fixes to Skiddy Toast, it removed the require for that native. I asked him to add it into his PR because I didn’t want to make it hard for anyone. Except he never finished it. I went ahead and fixed other things.

Reply:
At least, inside the RCS hook there was some issues I overlooked.

Reply:
If I understand what you're saying I should finish this work but not implement the fix with the native yet.

And wait for SkiddyToast to fix that?